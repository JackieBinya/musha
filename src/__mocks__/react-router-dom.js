module.exports = {
  ...jest.requireActual('react-router-dom'),
  useRouteMatch: jest.fn(),
  useLocation: jest.fn(),
};
