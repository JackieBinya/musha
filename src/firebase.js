import firebase from 'firebase/app';
import 'firebase/auth';
import 'firebase/firestore';
import 'firebase/storage';

const firebaseConfig = firebase.initializeApp({
  apiKey: 'AIzaSyDD14Pef51sWG0czExRugPOLJO-Ne44bNI',
  authDomain: 'musha-000.firebaseapp.com',
  databaseURL: 'https://musha-000.firebaseio.com',
  projectId: 'musha-000',
  storageBucket: 'musha-000.appspot.com',
  messagingSenderId: '111803589651',
  appId: '1:111803589651:web:b90b953a233696281b4be6',
});

const storage = firebase.storage();

export { storage, firebaseConfig as firebase };
