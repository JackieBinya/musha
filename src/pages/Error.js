import React from 'react';

export const Error = () => <h2>Error</h2>;