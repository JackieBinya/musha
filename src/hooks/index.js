import React, { useState, useEffect } from 'react';
import { firebase } from '../firebase';

export const useProperties = () => {
  const [properties, setProperties] = useState([]);

  useEffect(() => {
    firebase
      .firestore()
      .collection('properties')
      .get()
      .then((data) => {
        const allProperties = data.docs.map((doc) => ({
          ...doc.data(),
          id: doc.id,
        }));

        setProperties([...allProperties]);
      });
  }, []);

  return { properties, setProperties };
};

export const useProperty = (id) => {
  const [property, setProperty] = useState('');

  useEffect(() => {
    firebase
      .firestore()
      .collection('properties')
      .doc(id)
      .get()
      .then((doc) => {
        if (doc.exists) {
          const newProperty = doc.data();
          setProperty(newProperty);
        }
      })
      .catch((error) => console.log(error));
  }, []);

  return { property };
};

export const usePropertiesByUserID = (userId) => {
  const [userProperties, setUserProperties] = useState([]);
  console.log(userProperties);

  useEffect(() => {
    firebase
      .firestore()
      .collection('properties')
      .where('ownerID', '==', userId)
      .get()
      .then((data) => {
        const newProperties = data.docs.map((doc) => ({
          ...doc.data(),
          id: doc.id,
        }));

        setUserProperties([...newProperties]);
      })
      .catch((error) => console.log(error));
  }, []);

  return { userProperties };
};
