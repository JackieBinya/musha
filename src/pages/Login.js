import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import { SignUp } from './SignUp';
import { FormMessage } from '../components/FormMessage';
import { PasswordInput } from '../components/PasswordInput';
import { EmailInput } from '../components/EmailInput';
import { firebase } from '../firebase';

export const Login = ({ history }) => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [message, setMessage] = useState('');
  const [isAuthenticated, setIsAuthenticated] = useState(false);

  const handleLogin = async (e) => {
    e.preventDefault();

    try {
      await firebase.auth().signInWithEmailAndPassword(email, password);
      setIsAuthenticated(true);
      setMessage('Congrats 🥳, you are successfully logged in.');
      setTimeout(() => history.push('/my-properties'), 5000);
    } catch (error) {
      setIsAuthenticated(false);

      if (error.code === ' auth/invalid-email') {
        setMessage('🤖 email address is not valid');
      }
      if (error.code === 'auth/user-disabled') {
        setMessage('🤖 your account has been suspended');
      }
      if (error.code === 'auth/user-not-found') {
        setMessage('🤖 email does no exist!');
      }
      if (error.code === ' auth/wrong-password') {
        setMessage('🤖 incorrect password');
      }
    }
  };
  return (
    <>
      <Link to="/signup">Sign Up</Link>
      <h1>Login</h1>
      <form className="auth-form" onSubmit={handleLogin}>
        <FormMessage
          isAuthenticated={isAuthenticated}
          message={message}
          setMessage={setMessage}
        />
        <EmailInput email={email} setEmail={setEmail} />
        <PasswordInput setPassword={setPassword} password={password} />
        <button type="submit">Continue</button>
      </form>
      <Link to="/password-reset">Have you forgotten your password?</Link>
    </>
  );
};
