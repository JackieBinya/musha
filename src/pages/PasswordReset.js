import React, { useState } from 'react';
import { firebase } from '../firebase';
import { EmailInput } from '../components/EmailInput';
import { AlertMessage } from '../components/AlertMessage';

/********************
 *  Add continue url : https://firebase.google.com/docs/auth/web/passing-state-in-email-actions#passing_statecontinue_url_in_email_actions
 */

export const PasswordReset = ({ history }) => {
  const [email, setEmail] = useState('');
  const [message, setMessage] = useState('');
  const [isEmailSent, setIsEmailSent] = useState(false);

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      await firebase.auth().sendPasswordResetEmail(email);
      setIsEmailSent(true);
      setMessage(
        'Check your inbox for further instructions on how to reset your password! '
      );
      setTimeout(() => history.push('/'), 5000);
    } catch (error) {
      setIsEmailSent(false);
      if (error.code === 'auth/invalid-email') {
        setMessage('Email invalid');
      }

      if (error.code === 'auth/user-not-found') {
        setMessage(
          'An account associated with the provided email does not exist!'
        );
      }
    }
  };

  return (
    <div className="container">
      <div className="container-inner">
        <div className="form-container">
          <AlertMessage
            isAuthenticated={isEmailSent}
            message={message}
            setMessage={setMessage}
          />
          <h1>Password Reset</h1>
          <form onSubmit={handleSubmit} className="form-primary">
            <EmailInput email={email} setEmail={setEmail} />
            <button data-testid="submit-action" type="submit">
              Reset my password
            </button>
          </form>
        </div>
      </div>
    </div>
  );
};
