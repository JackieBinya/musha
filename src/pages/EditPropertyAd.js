import React, { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { useForm } from 'react-hook-form';
import { firebase } from '../firebase';
import { yupResolver } from '@hookform/resolvers';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faChevronLeft } from '@fortawesome/free-solid-svg-icons';

import { useProperty } from '../hooks';
import { FirstSection } from '../components/FirstSection';
import { LocationSection } from '../components/LocationSection';
import { Loader } from '../components/Loader';
import { PropertyDetailsSection } from '../components/PropertyDetailsSection';
import { PROPERTY_SCHEMA } from '../constants/validations';

export const EditPropertyAds = ({ history }) => {
  const { propertyId } = useParams();
  const { property } = useProperty(propertyId);

  const { register, handleSubmit, errors } = useForm({
    resolver: yupResolver(PROPERTY_SCHEMA),
  });

  const [isTitleUpdated, setIsTitleUpdated] = useState(false);
  const [isMobileNumberUpdated, setIsMobileNumberUpdated] = useState(false);
  const [isAvailableToUpdated, setIsAvailableToUpdated] = useState(false);
  const [isCityUpdated, setIsCityUpdated] = useState(false);
  const [isLocationUpdated, setIsLocationUpdated] = useState(false);
  const [isNumberOfBathroomsUpdated, setIsNumberOfBathroomsUpdated] = useState(
    false
  );
  const [isNumberOfBedroomsUpdated, setIsNumberOfBedroomsUpdated] = useState(
    false
  );
  const [isDescriptionUpdated, setIsDescriptionUpdated] = useState(false);

  const [isModalVisible, setIsModalVisible] = useState(false);

  const onSubmit = (
    {
      city,
      location,
      availableTo,
      numberOfBathrooms,
      numberOfBedrooms,
      description,
      mobileNumber,
      title,
    },
    e
  ) => {
    const propertyRef = firebase
      .firestore()
      .collection('properties')
      .doc(propertyId);

    if (property.title !== title) {
      propertyRef
        .update({
          title,
        })
        .then(() => setIsTitleUpdated(true))
        .catch((error) => console.log(error));
    }

    if (property.availableTo !== availableTo) {
      propertyRef
        .update({
          availableTo,
        })
        .then(() => setIsAvailableToUpdated(true))
        .catch((error) => console.log(error));
    }

    if (property.mobileNumber !== mobileNumber) {
      propertyRef
        .update({
          mobileNumber,
        })
        .then(() => setIsMobileNumberUpdated(true))
        .catch((error) => console.log(error));
    }

    if (property.city !== city) {
      propertyRef
        .update({
          city,
        })
        .then(() => setIsCityUpdated(true))
        .catch((error) => console.log(error));
    }

    if (property.location !== location) {
      propertyRef
        .update({
          location,
        })
        .then(() => setIsLocationUpdated(true))
        .catch((error) => console.log(error));
    }

    if (property.description !== description) {
      propertyRef
        .update({
          description,
        })
        .then(() => setIsDescriptionUpdated(true))
        .catch((error) => console.log(error));
    }

    if (property.numberOfBedrooms !== numberOfBedrooms) {
      propertyRef
        .update({
          numberOfBedrooms,
        })
        .then(() => setIsNumberOfBedroomsUpdated(true))
        .catch((error) => console.log(error));
    }

    if (property.numberOfBathrooms !== numberOfBathrooms) {
      propertyRef
        .update({
          numberOfBathrooms,
        })
        .then(() => setIsNumberOfBathroomsUpdated(true))
        .catch((error) => console.log(error));
    }
  };

  useEffect(() => {
    if (
      isTitleUpdated ||
      isMobileNumberUpdated ||
      isAvailableToUpdated ||
      isCityUpdated ||
      isLocationUpdated ||
      isNumberOfBathroomsUpdated ||
      isNumberOfBedroomsUpdated ||
      isDescriptionUpdated
    ) {
      setIsModalVisible(true);

      setIsTitleUpdated(false);
      setIsAvailableToUpdated(false);
      setIsMobileNumberUpdated(false);
      setIsCityUpdated(false);
      setIsLocationUpdated(false);
      setIsNumberOfBathroomsUpdated(false);
      setIsNumberOfBedroomsUpdated(false);
      setIsDescriptionUpdated(false);
    }
  }, [
    isTitleUpdated,
    isAvailableToUpdated,
    isCityUpdated,
    isLocationUpdated,
    isMobileNumberUpdated,
    isDescriptionUpdated,
    isNumberOfBathroomsUpdated,
    isNumberOfBedroomsUpdated,
  ]);

  return (
    <div className="container">
      <div style={{ marginBottom: '2em' }}>
        {property && isModalVisible && (
          <div className="form-modal-container">
            <p
              className={isModalVisible ? 'show' : 'hide'}
            >{`Property ID:${propertyId} is successfully updated`}</p>
            <button
              data-testid="go-to-my-properties-action"
              type="button"
              onClick={() => history.goBack()}
            >
              {' '}
              Go to my properties
            </button>
          </div>
        )}
        <div className="property-form-container">
          {!property ? (
            <Loader />
          ) : (
            <div
              data-testid="edit-form"
              className={isModalVisible ? 'hide' : 'show'}
            >
              <button
                data-testid="back-button"
                type="button"
                onClick={() => history.goBack()}
              >
                <FontAwesomeIcon
                  icon={faChevronLeft}
                  className="property-icons-svg"
                />
                Back
              </button>

              <h1>Edit property ad</h1>

              <form onSubmit={handleSubmit(onSubmit)} className="form-primary">
                <FirstSection
                  defaultMobileNumber={property.mobileNumber}
                  defaultTitle={property.title}
                  defaultAvailableTo={property.availableTo}
                  register={register}
                  errors={errors}
                />

                <LocationSection
                  defaultCity={property.city}
                  defaultLocation={property.location}
                  register={register}
                  errors={errors}
                />

                {/* <UploadImagesSection
                  defaultImageUrls={property.imageUrls}
                  setImageUrls={setImageUrls}
                /> */}

                <PropertyDetailsSection
                  defaultNumberOfBathrooms={property.numberOfBathrooms}
                  defaultNumberOfBedrooms={property.numberOfBedrooms}
                  defaultDescription={property.description}
                  register={register}
                  errors={errors}
                />

                <button data-testid="submit-action" type="submit">
                  Save
                </button>
              </form>
            </div>
          )}
        </div>
      </div>
    </div>
  );
};
