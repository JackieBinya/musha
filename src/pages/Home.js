import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import Fuse from 'fuse.js';
import { useProperties } from '../hooks';
import { ShortProperty } from '../components/ShortProperty';
import { Hero } from '../components/Hero';
import { Loader } from '../components/Loader';
import { PropertyIconsHome } from '../components/PropertyIconsHome';

export const Home = () => {
  const { properties } = useProperties();
  const [query, setQuery] = useState('');

  const fuse = new Fuse(properties, {
    keys: ['location', 'city'],
    includeScore: true,
  });

  const results = fuse.search(query);
  const propertiesResults = query
    ? results.map((character) => character.item)
    : properties;

  return (
    <>
      <Hero query={query} setQuery={setQuery} />
      <main>
        <div className="main container">
          {propertiesResults.length === 0 ? (
            <Loader />
          ) : propertiesResults ? (
            propertiesResults.map(
              ({
                id,
                location,
                imageUrls,
                numberOfBathrooms,
                numberOfBedrooms,
                city,
                title,
                description,
              }) => (
                <li key={id} className="main-list-item">
                  <Link to={`/property/${id}`} className="main-link">
                    <ShortProperty
                      imageUrls={imageUrls}
                      title={title}
                      location={location}
                      description={description}
                    >
                      <PropertyIconsHome
                        city={city}
                        title={title}
                        location={location}
                        numberOfBathrooms={numberOfBathrooms}
                        numberOfBedrooms={numberOfBedrooms}
                      />
                    </ShortProperty>
                  </Link>
                </li>
              )
            )
          ) : (
            ''
          )}
        </div>
      </main>
    </>
  );
};
