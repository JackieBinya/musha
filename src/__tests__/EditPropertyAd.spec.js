import React from 'react';
import { firebase } from '../firebase';
import { useProperty } from '../hooks';
import { useParams } from 'react-router-dom';
import {
  render,
  cleanup,
  fireEvent,
  act,
  waitFor,
} from '@testing-library/react';
import { EditPropertyAds } from '../pages/EditPropertyAd';

import { toHaveClass } from '@testing-library/jest-dom/matchers';

expect.extend({ toHaveClass });

jest.mock('../hooks', () => ({
  useProperty: jest.fn(),
}));

jest.mock('../firebase', () => ({
  firebase: {
    firestore: jest.fn(() => ({
      collection: jest.fn((properties) => ({
        doc: jest.fn((id) => ({
          update: jest.fn((data) => {
            return new Promise((resolve, reject) => {
              if (!data) reject();

              resolve();
            });
          }),
        })),
      })),
    })),
  },
}));

beforeEach(cleanup);

describe('<EditPropertyAd', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('The loader component is rendered during API call to fetch resources.', () => {
    useParams.mockReturnValue({ propertyId: '1' });
    useProperty.mockReturnValue({ property: null });

    const { queryByText } = render(<EditPropertyAds />);

    expect(queryByText('Loading...')).toBeTruthy();
  });

  it('The component is successfully rendered when the property resource is fetched from the server.', () => {
    useParams.mockReturnValue({ propertyId: '1' });
    useProperty.mockReturnValue({
      property: {
        city: 'Test City',
        location: 'Test Location',
        availableTo: 'rent',
        numberOfBathrooms: '2',
        numberOfBedrooms: '3',
        description: 'Test Description',
        mobileNumber: '123456',
        title: 'Test Title',
      },
    });

    const { asFragment } = render(<EditPropertyAds />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('Back button action is captured', () => {
    useParams.mockReturnValue({ propertyId: '1' });
    useProperty.mockReturnValue({
      property: {
        city: 'Test City',
        location: 'Test Location',
        availableTo: 'rent',
        numberOfBathrooms: '2',
        numberOfBedrooms: '3',
        description: 'Test Description',
        mobileNumber: '123456',
        title: 'Test Title',
      },
    });

    const history = {
      goBack: jest.fn(),
    };

    const { queryByTestId } = render(<EditPropertyAds history={history} />);

    const backButton = queryByTestId('back-button');

    fireEvent.click(backButton);

    expect(history.goBack).toBeCalledTimes(1);
  });
});

describe('The user can update a property ad', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  useParams.mockReturnValue({ propertyId: '1' });
  useProperty.mockReturnValue({
    property: {
      city: 'Test City',
      location: 'Test Location',
      availableTo: 'rent',
      numberOfBathrooms: '2',
      numberOfBedrooms: '3',
      description: 'Test Description',
      mobileNumber: '123456',
      title: 'Test Title',
    },
  });

  it('User can update the city', async () => {
    const { queryByText, queryByLabelText, queryByTestId } = render(
      <EditPropertyAds />
    );

    const cityInput = queryByLabelText('*City:', { selector: 'input' });

    const submitAction = queryByTestId('submit-action');

    act(() => {
      fireEvent.change(cityInput, { target: { value: 'Updated City' } });
    });

    fireEvent.submit(submitAction);

    await waitFor(() => {
      expect(
        queryByText('Property ID:1 is successfully updated.')
      ).toBeTruthy();
      expect(queryByTestId('edit-form')).toHaveClass('hide');
    });
  });

  it('User can update the location', async () => {
    const { queryByText, queryByLabelText, queryByTestId } = render(
      <EditPropertyAds />
    );

    const locationInput = queryByLabelText('*Location/Suburb:', {
      selector: 'input',
    });

    const submitAction = queryByTestId('submit-action');

    act(() => {
      fireEvent.change(locationInput, {
        target: { value: 'Updated Location' },
      });
    });

    fireEvent.submit(submitAction);

    await waitFor(() => {
      expect(
        queryByText('Property ID:1 is successfully updated.')
      ).toBeTruthy();
      expect(queryByTestId('edit-form')).toHaveClass('hide');
    });
  });

  it('User can update the mobile number', async () => {
    const { queryByText, queryByLabelText, queryByTestId } = render(
      <EditPropertyAds />
    );

    const mobileNumberInput = queryByLabelText('*Your mobile number:', {
      selector: 'input',
    });

    const submitAction = queryByTestId('submit-action');

    act(() => {
      fireEvent.change(mobileNumberInput, { target: { value: '000111' } });
    });

    fireEvent.submit(submitAction);

    await waitFor(() => {
      expect(
        queryByText('Property ID:1 is successfully updated.')
      ).toBeTruthy();
      expect(queryByTestId('edit-form')).toHaveClass('hide');
    });
  });

  it('User can update the available to field', async () => {
    const { queryByText, queryByLabelText, queryByTestId } = render(
      <EditPropertyAds />
    );

    const availableForSale = queryByLabelText('Sale');

    const submitAction = queryByTestId('submit-action');

    act(() => {
      fireEvent.click(availableForSale);
    });

    fireEvent.submit(submitAction);

    await waitFor(() => {
      expect(
        queryByText('Property ID:1 is successfully updated.')
      ).toBeTruthy();
      expect(queryByTestId('edit-form')).toHaveClass('hide');
    });
  });

  it('User can update the mobile number', async () => {
    const { queryByText, queryByLabelText, queryByTestId } = render(
      <EditPropertyAds />
    );

    const mobileNumberInput = queryByLabelText('*Your mobile number:', {
      selector: 'input',
    });

    const submitAction = queryByTestId('submit-action');

    act(() => {
      fireEvent.change(mobileNumberInput, {
        target: { value: 'Updated Title' },
      });
    });

    fireEvent.submit(submitAction);

    await waitFor(() => {
      expect(
        queryByText('Property ID:1 is successfully updated.')
      ).toBeTruthy();
      expect(queryByTestId('edit-form')).toHaveClass('hide');
    });
  });

  it('User can update the number of bathrooms', async () => {
    const { queryByText, queryByLabelText, queryByTestId } = render(
      <EditPropertyAds />
    );

    const selectBathrooms = queryByLabelText(
      'Choose the number of bathrooms in your property:'
    );

    const submitAction = queryByTestId('submit-action');

    act(() => {
      fireEvent.change(selectBathrooms, { target: { value: 'studio' } });
    });

    fireEvent.submit(submitAction);

    await waitFor(() => {
      expect(
        queryByText('Property ID:1 is successfully updated.')
      ).toBeTruthy();
      expect(queryByTestId('edit-form')).toHaveClass('hide');
    });
  });

  it('User can update the number of bedrooms and the go-to-my-properties-action can be captured.', async () => {
    const history = {
      goBack: jest.fn(),
    };

    const { queryByText, queryByLabelText, queryByTestId } = render(
      <EditPropertyAds history={history} />
    );

    const selectBedrooms = queryByLabelText(
      'Choose the number of bedrooms in your property:'
    );

    const submitAction = queryByTestId('submit-action');

    act(() => {
      fireEvent.change(selectBedrooms, { target: { value: 'shared' } });
    });

    fireEvent.submit(submitAction);

    await waitFor(() => {
      expect(
        queryByText('Property ID:1 is successfully updated.')
      ).toBeTruthy();
      expect(queryByTestId('edit-form')).toHaveClass('hide');
    });

    const goToMyPropertiesAction = queryByTestId('go-to-my-properties-action');

    fireEvent.click(goToMyPropertiesAction);

    expect(history.goBack).toBeCalledTimes(1);
  });
});
