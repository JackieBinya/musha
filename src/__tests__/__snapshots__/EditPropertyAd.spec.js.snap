// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditPropertyAd The component is successfully rendered when the property resource is fetched from the server. 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="form-wrapper"
    >
      <div
        style="margin-bottom: 2em;"
      >
        <div
          class="property-form-container"
        >
          <div
            class="show"
            data-testid="edit-form"
          >
            <button
              data-testid="back-button"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-chevron-left fa-w-10 property-icons-svg"
                data-icon="chevron-left"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 320 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
                  fill="currentColor"
                />
              </svg>
              Back
            </button>
            <h1>
              Edit property ad
            </h1>
            <form
              class="form-primary"
            >
              <section
                class="step-wrapper"
              >
                <h3
                  class="step-headliner"
                >
                  Miscellaneous Details
                </h3>
                <div
                  class="step-contents"
                >
                  <p>
                    *The property is available for :
                  </p>
                  <div
                    class="radio-wrapper"
                  >
                    <div>
                      <label>
                        <input
                          checked=""
                          name="availableTo"
                          type="radio"
                          value="rent"
                        />
                        Rent
                      </label>
                    </div>
                    <div>
                      <label>
                        <input
                          name="availableTo"
                          type="radio"
                          value="sale"
                        />
                        Sale
                      </label>
                    </div>
                  </div>
                  <div>
                    <label
                      for="title"
                    >
                      *Ad title:
                    </label>
                    <input
                      id="title"
                      name="title"
                      type="text"
                      value="Test Title"
                    />
                  </div>
                  <div>
                    <label
                      for="mobileNumber"
                    >
                      *Your mobile number:
                    </label>
                    <input
                      id="mobileNumber"
                      name="mobileNumber"
                      type="text"
                      value="123456"
                    />
                  </div>
                </div>
              </section>
              <section
                class="step-wrapper"
              >
                <h3
                  class="step-headliner"
                >
                  Location
                </h3>
                <div
                  class="step-contents"
                >
                  <div>
                    <label
                      for="city"
                    >
                      *City:
                    </label>
                    <input
                      id="city"
                      name="city"
                      type="text"
                      value="Test City"
                    />
                  </div>
                  <div>
                    <label
                      for="location"
                    >
                      *Location/Suburb:
                    </label>
                    <input
                      id="location"
                      name="location"
                      type="text"
                      value="Test Location"
                    />
                  </div>
                </div>
              </section>
              <section
                class="step-wrapper"
              >
                <h3
                  class="step-headliner"
                >
                  Property Details
                </h3>
                <div
                  class="step-contents"
                >
                  <div
                    class="step-details-wrapper"
                  >
                    <label>
                      Choose the number of bedrooms in your property:
                      <select
                        name="numberOfBedrooms"
                      >
                        <option
                          value="studio"
                        >
                          Studio
                        </option>
                        <option
                          value="1"
                        >
                          1 bedroom
                        </option>
                        <option
                          value="2"
                        >
                          2 bedrooms
                        </option>
                        <option
                          selected=""
                          value="3"
                        >
                          3+ bedrooms
                        </option>
                      </select>
                    </label>
                  </div>
                  <div
                    class="step-details-wrapper"
                  >
                    <label>
                      Choose the number of bathrooms in your property:
                      <select
                        name="numberOfBathrooms"
                      >
                        <option
                          value="shared"
                        >
                          shared bathroom
                        </option>
                        <option
                          value="1"
                        >
                          1 bathroom
                        </option>
                        <option
                          selected=""
                          value="2"
                        >
                          2 bathrooms
                        </option>
                        <option
                          value="3"
                        >
                          3+ bathrooms
                        </option>
                      </select>
                    </label>
                  </div>
                  <div
                    class="step-details-wrapper"
                  >
                    <label
                      for="description"
                    >
                      *Description:
                    </label>
                    <textarea
                      id="description"
                      name="description"
                      placeholder="Type your description here..."
                    >
                      Test Description
                    </textarea>
                  </div>
                </div>
              </section>
              <button
                data-testid="submit-action"
                type="submit"
              >
                Save
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
